Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/asyncio/base_events.py", line 650, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/contextlib.py", line 222, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/client.py", line 608, in async_setup_kernel
    yield
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
data = pd.read_excel('bundesliga_top7_offensive.xlsx', index_col=0)
data.head(5)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/pandas/compat/_optional.py:132[0m, in [0;36mimport_optional_dependency[0;34m(name, extra, errors, min_version)[0m
[1;32m    131[0m [38;5;28;01mtry[39;00m:
[0;32m--> 132[0m     module [38;5;241m=[39m importlib[38;5;241m.[39mimport_module(name)
[1;32m    133[0m [38;5;28;01mexcept[39;00m [38;5;167;01mImportError[39;00m:

File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/importlib/__init__.py:126[0m, in [0;36mimport_module[0;34m(name, package)[0m
[1;32m    125[0m         level [38;5;241m+[39m[38;5;241m=[39m [38;5;241m1[39m
[0;32m--> 126[0m [38;5;28;01mreturn[39;00m _bootstrap[38;5;241m.[39m_gcd_import(name[level:], package, level)

File [0;32m<frozen importlib._bootstrap>:1206[0m, in [0;36m_gcd_import[0;34m(name, package, level)[0m

File [0;32m<frozen importlib._bootstrap>:1178[0m, in [0;36m_find_and_load[0;34m(name, import_)[0m

File [0;32m<frozen importlib._bootstrap>:1142[0m, in [0;36m_find_and_load_unlocked[0;34m(name, import_)[0m

[0;31mModuleNotFoundError[0m: No module named 'openpyxl'

During handling of the above exception, another exception occurred:

[0;31mImportError[0m                               Traceback (most recent call last)
Cell [0;32mIn[11], line 1[0m
[0;32m----> 1[0m data [38;5;241m=[39m pd[38;5;241m.[39mread_excel([38;5;124m'[39m[38;5;124mbundesliga_top7_offensive.xlsx[39m[38;5;124m'[39m, index_col[38;5;241m=[39m[38;5;241m0[39m)
[1;32m      2[0m data[38;5;241m.[39mhead([38;5;241m5[39m)

File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/pandas/io/excel/_base.py:504[0m, in [0;36mread_excel[0;34m(io, sheet_name, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, parse_dates, date_parser, date_format, thousands, decimal, comment, skipfooter, storage_options, dtype_backend, engine_kwargs)[0m
[1;32m    502[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [38;5;28misinstance[39m(io, ExcelFile):
[1;32m    503[0m     should_close [38;5;241m=[39m [38;5;28;01mTrue[39;00m
[0;32m--> 504[0m     io [38;5;241m=[39m ExcelFile(
[1;32m    505[0m         io,
[1;32m    506[0m         storage_options[38;5;241m=[39mstorage_options,
[1;32m    507[0m         engine[38;5;241m=[39mengine,
[1;32m    508[0m         engine_kwargs[38;5;241m=[39mengine_kwargs,
[1;32m    509[0m     )
[1;32m    510[0m [38;5;28;01melif[39;00m engine [38;5;129;01mand[39;00m engine [38;5;241m!=[39m io[38;5;241m.[39mengine:
[1;32m    511[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m    512[0m         [38;5;124m"[39m[38;5;124mEngine should not be specified when passing [39m[38;5;124m"[39m
[1;32m    513[0m         [38;5;124m"[39m[38;5;124man ExcelFile - ExcelFile already has the engine set[39m[38;5;124m"[39m
[1;32m    514[0m     )

File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/pandas/io/excel/_base.py:1580[0m, in [0;36mExcelFile.__init__[0;34m(self, path_or_buffer, engine, storage_options, engine_kwargs)[0m
[1;32m   1577[0m [38;5;28mself[39m[38;5;241m.[39mengine [38;5;241m=[39m engine
[1;32m   1578[0m [38;5;28mself[39m[38;5;241m.[39mstorage_options [38;5;241m=[39m storage_options
[0;32m-> 1580[0m [38;5;28mself[39m[38;5;241m.[39m_reader [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_engines[engine](
[1;32m   1581[0m     [38;5;28mself[39m[38;5;241m.[39m_io,
[1;32m   1582[0m     storage_options[38;5;241m=[39mstorage_options,
[1;32m   1583[0m     engine_kwargs[38;5;241m=[39mengine_kwargs,
[1;32m   1584[0m )

File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/pandas/io/excel/_openpyxl.py:552[0m, in [0;36mOpenpyxlReader.__init__[0;34m(self, filepath_or_buffer, storage_options, engine_kwargs)[0m
[1;32m    534[0m [38;5;129m@doc[39m(storage_options[38;5;241m=[39m_shared_docs[[38;5;124m"[39m[38;5;124mstorage_options[39m[38;5;124m"[39m])
[1;32m    535[0m [38;5;28;01mdef[39;00m [38;5;21m__init__[39m(
[1;32m    536[0m     [38;5;28mself[39m,
[0;32m   (...)[0m
[1;32m    539[0m     engine_kwargs: [38;5;28mdict[39m [38;5;241m|[39m [38;5;28;01mNone[39;00m [38;5;241m=[39m [38;5;28;01mNone[39;00m,
[1;32m    540[0m ) [38;5;241m-[39m[38;5;241m>[39m [38;5;28;01mNone[39;00m:
[1;32m    541[0m [38;5;250m    [39m[38;5;124;03m"""[39;00m
[1;32m    542[0m [38;5;124;03m    Reader using openpyxl engine.[39;00m
[1;32m    543[0m 
[0;32m   (...)[0m
[1;32m    550[0m [38;5;124;03m        Arbitrary keyword arguments passed to excel engine.[39;00m
[1;32m    551[0m [38;5;124;03m    """[39;00m
[0;32m--> 552[0m     import_optional_dependency([38;5;124m"[39m[38;5;124mopenpyxl[39m[38;5;124m"[39m)
[1;32m    553[0m     [38;5;28msuper[39m()[38;5;241m.[39m[38;5;21m__init__[39m(
[1;32m    554[0m         filepath_or_buffer,
[1;32m    555[0m         storage_options[38;5;241m=[39mstorage_options,
[1;32m    556[0m         engine_kwargs[38;5;241m=[39mengine_kwargs,
[1;32m    557[0m     )

File [0;32m/opt/homebrew/Caskroom/miniconda/base/envs/python311/lib/python3.11/site-packages/pandas/compat/_optional.py:135[0m, in [0;36mimport_optional_dependency[0;34m(name, extra, errors, min_version)[0m
[1;32m    133[0m [38;5;28;01mexcept[39;00m [38;5;167;01mImportError[39;00m:
[1;32m    134[0m     [38;5;28;01mif[39;00m errors [38;5;241m==[39m [38;5;124m"[39m[38;5;124mraise[39m[38;5;124m"[39m:
[0;32m--> 135[0m         [38;5;28;01mraise[39;00m [38;5;167;01mImportError[39;00m(msg)
[1;32m    136[0m     [38;5;28;01mreturn[39;00m [38;5;28;01mNone[39;00m
[1;32m    138[0m [38;5;66;03m# Handle submodules: if we have submodule, grab parent module from sys.modules[39;00m

[0;31mImportError[0m: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.
ImportError: Missing optional dependency 'openpyxl'.  Use pip or conda to install openpyxl.

